include/xframe/xaxis.hpp

```diff
@@ -56,28 +60,27 @@ namespace xf
      *********/
 
     template <class L, class T, class MT = hash_map_tag>
-    class xaxis
+    class xaxis : public xaxis_base<xaxis<L, T, MT>>
     {
     public:
 
-        static_assert(std::is_integral<T>::value, "mapped_type T must be an integral type");
-
+        using base_type = xaxis_base<xaxis>;
         using self_type = xaxis<L, T, MT>;
-        using label_list = std::vector<L>;
-        using key_type = L;
-        using mapped_type = T;
+        using key_type = typename base_type::key_type;
+        using label_list = std::vector<key_type>;
```

Comment:
`label_list` should be defined as `typename base_type::label_list` so it is consistent with the definitions of `size_type` and `difference_type` below.